getting-started/authenticating-users/app.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The authenticating-users program is a sample web server application that
+// extracts and verifies user identity data passed to it via Identity-Aware
+// Proxy.
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+    "net/http"
+	"os"
+
+	"cloud.google.com/go/compute/metadata"
+	"github.com/dgrijalva/jwt-go"
+)
+
+var cachedCertificates = make(map[string]string)
+
+func main() {
+	http.HandleFunc("/", indexHandler)
+
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = "8080"
+		log.Printf("Defaulting to port %s", port)
+	}
+
+	log.Printf("Listening on port %s", port)
+	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%s", port), nil))
+}
+
+func certs() (map[string]string, error) {
+	const url = "https://www.gstatic.com/iap/verify/public_key"
+
+	if len(cachedCertificates) != 0 { // Already got them previously
+		return cachedCertificates, nil
+	}
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return cachedCertificates, err
+	}
+
+	dec := json.NewDecoder(resp.Body)
+	err = dec.Decode(&cachedCertificates)
+	if err != nil {
+		return cachedCertificates, err
+	}
+
+	return cachedCertificates, nil
+}
+
+func audience() (string, error) {
+	client := metadata.NewClient(&http.Client{Transport: userAgentTransport{
+		userAgent: "my-user-agent",
+		base:      http.DefaultTransport,
+	}})
```

Comment:
I've only used the default metadata client before. Does this need to include a user agent? If not, you can just do `metadata.NumericProjectID()`.

Reply:
I tried that, and it didn't work. I tried a lot of things, and ended up having to copy the only example I could find verbatim. It seems like it shouldn't need to be that complicated, though.

Reply:
Where were you running it?

@broady, any ideas why this would be needed?

Reply:
I tried this without the user agent and it seems to work.

```
package main

import (
	"fmt"
	"net/http"
	"os"

	"cloud.google.com/go/compute/metadata"
)

func main() {
	port := os.Getenv("PORT")
	if port == "" {
		port = "8080"
	}
	http.HandleFunc("/", index)
	http.ListenAndServe(":"+port, nil)
}

func index(w http.ResponseWriter, r *http.Request) {
	projectID, err := metadata.ProjectID()
	if err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}
	projectNumber, err := metadata.NumericProjectID()
	if err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
	}
	fmt.Fprintf(w, "Hello, %v (%v)!", projectID, projectNumber)
}
```

Reply:
It worked for me, too, so I made the change. There must have been something else going on when I was having the problem.